app/src/main/java/com/marverenic/music/utils/SlsInteraction.java

```diff
@@ -0,0 +1,53 @@
+package com.marverenic.music.utils;
+
+import android.content.Context;
+import android.content.Intent;
+
+import com.marverenic.music.model.Song;
+
+public class SlsInteraction {
+    private static final int START = 0;
+    private static final int RESUME = 1;
+    private static final int PAUSE = 2;
+    private static final int COMPLETE = 3;
+
+    private static final String SLS_ACTION = "com.adam.aslfms.notify.playstatechanged";
+
+    private Context mContext;
+
+    public SlsInteraction(Context context) {
+        mContext = context;
+    }
+
+    private Intent base() {
+        String appName = mContext.getApplicationInfo().loadLabel(mContext.getPackageManager()).toString();
+        return new Intent(SLS_ACTION)
+                .putExtra("app-name", appName)
+                .putExtra("app-package", mContext.getPackageName());
+    }
+
+    private Intent withSong(Song song) {
+        return base()
```

Comment:
That's a good point – I hadn't thought about async issues because was in the mindset of immutable objects and builders. This should be fine as-is then, since there isn't really any performance hit for allocating Intent objects here.